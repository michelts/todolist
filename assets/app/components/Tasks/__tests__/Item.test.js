describe('TaskDetail component', () => {
  it('should render the task description, a button to disable it, the date, and a priority field', () => {
  });
});
