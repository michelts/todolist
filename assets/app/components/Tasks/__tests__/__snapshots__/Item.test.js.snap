// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks component should render the proper priority option selected: option 3 selected 1`] = `
<FormControl
  as="select"
  custom={true}
  name="priority"
  onChange={[Function]}
  size="sm"
>
  <option
    key="0"
    selected={false}
    value={0}
  >
    0
  </option>
  <option
    key="1"
    selected={false}
    value={1}
  >
    1
  </option>
  <option
    key="2"
    selected={false}
    value={2}
  >
    2
  </option>
  <option
    key="3"
    selected={true}
    value={3}
  >
    3
  </option>
  <option
    key="4"
    selected={false}
    value={4}
  >
    4
  </option>
</FormControl>
`;

exports[`Tasks component should render the task description, a button to select it, the date, and a priority field 1`] = `
<ListGroupItem
  active={false}
  disabled={false}
  variant={null}
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <div
      className="w-10 text-center"
    >
      <div>
        Done?
      </div>
      <FormCheck
        checked={false}
        disabled={false}
        inline={false}
        isInvalid={false}
        isValid={false}
        name="is-completed"
        onClick={[Function]}
        title=""
        type="checkbox"
      />
    </div>
    <div
      className="flex-grow-1 ml-4"
    >
      <EditText
        data-name="description"
        onSave={[Function]}
        value="Task 1"
      />
      <span
        className="text-muted"
      >
        <EditText
          data-name="due-date"
          onSave={[Function]}
          value="2020-05-22"
        />
      </span>
    </div>
    <div
      className="w-10 ml-4 text-center"
    >
      <div
        className="mb-1"
      >
        Priority
      </div>
      <FormControl
        as="select"
        custom={true}
        name="priority"
        onChange={[Function]}
        size="sm"
      >
        <option
          key="0"
          selected={true}
          value={0}
        >
          0
        </option>
        <option
          key="1"
          selected={false}
          value={1}
        >
          1
        </option>
        <option
          key="2"
          selected={false}
          value={2}
        >
          2
        </option>
        <option
          key="3"
          selected={false}
          value={3}
        >
          3
        </option>
        <option
          key="4"
          selected={false}
          value={4}
        >
          4
        </option>
      </FormControl>
    </div>
    <div
      className="w-10 ml-4"
    >
      <Button
        active={false}
        className="btn-sm"
        disabled={false}
        type="button"
        variant="danger"
      >
        Archive
      </Button>
    </div>
  </div>
</ListGroupItem>
`;

exports[`Tasks component should render the task description, a button to select it, the date, and a priority field 2`] = `
<ListGroupItem
  active={false}
  disabled={false}
  variant={null}
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <div
      className="w-10 text-center"
    >
      <div>
        Done?
      </div>
      <FormCheck
        checked={false}
        disabled={false}
        inline={false}
        isInvalid={false}
        isValid={false}
        name="is-completed"
        onClick={[Function]}
        title=""
        type="checkbox"
      />
    </div>
    <div
      className="flex-grow-1 ml-4"
    >
      <EditText
        data-name="description"
        onSave={[Function]}
        value="Fill with the task description"
      />
      <span
        className="text-muted"
      >
        <EditText
          data-name="due-date"
          onSave={[Function]}
          value="No date"
        />
      </span>
    </div>
    <div
      className="w-10 ml-4 text-center"
    >
      <div
        className="mb-1"
      >
        Priority
      </div>
      <FormControl
        as="select"
        custom={true}
        name="priority"
        onChange={[Function]}
        size="sm"
      >
        <option
          key="0"
          selected={true}
          value={0}
        >
          0
        </option>
        <option
          key="1"
          selected={false}
          value={1}
        >
          1
        </option>
        <option
          key="2"
          selected={false}
          value={2}
        >
          2
        </option>
        <option
          key="3"
          selected={false}
          value={3}
        >
          3
        </option>
        <option
          key="4"
          selected={false}
          value={4}
        >
          4
        </option>
      </FormControl>
    </div>
    <div
      className="w-10 ml-4"
    >
      <Button
        active={false}
        className="btn-sm"
        disabled={false}
        type="button"
        variant="danger"
      >
        Archive
      </Button>
    </div>
  </div>
</ListGroupItem>
`;
